app/controllers/maintenances_controller.rb

```diff
@@ -1,2 +1,69 @@
 class MaintenancesController < ApplicationController
+  before_action :set_maintenance, only: %i[show edit update destroy]
+  before_action :set_vehicle, only: %i[index create edit update]
+
+  def index
+    populate_index
+
+    @maintenance = Maintenance.new
+    @btn_txt = "Ajouter"
+  end
+
+  def create
+    @maintenance = Maintenance.new(maintenance_params)
+    @maintenance.vehicle_id = params[:vehicle_id]
+
+    if @maintenance.save
+      redirect_to vehicle_maintenance_operations_path(params[:vehicle_id], @maintenance)
```

Comment:
I think that you could use the instance variable @vehicle instead of digging into the params since you used a method to get the desired vehicle.
```suggestion
    @maintenance.vehicle_id = @vehicle

    if @maintenance.save
      redirect_to vehicle_maintenance_operations_path(@vehicle, @maintenance)
```